lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -801,9 +800,43 @@ def _histogram_binding_profile(self, n_time_bins, bandwidth, n_position_points):
 
         return x.squeeze(), densities
 
-    @use_docstring_from(KymoTrack.estimate_diffusion)
-    def estimate_diffusion(self, *args, **kwargs):
-        return [k.estimate_diffusion(*args, **kwargs) for k in self._src]
+    def estimate_diffusion(self, method, *args, min_length=None, **kwargs):
+        r"""Estimate diffusion constant for each track in the group.
+
+        Calls :meth:`KymoTrack.estimate_diffusion` for each track. See the documentation for that
+        method for more detailed information and references.
+
+        Parameters
+        ----------
+        method : {"cve", "ols", "gls"}
+            - "cve" : Covariance based estimator. Optimal if SNR > 1. Can only be used when
+              track is equidistantly sampled.
+            - "ols" : Ordinary least squares. Determines optimal number of lags.
+            - "gls" : Generalized least squares. Takes into account covariance matrix (slower).
+        max_lag : int (optional)
+            Number of lags to include when using an MSD-based estimator. When omitted, the method
+            will choose an appropriate number of lags to use. For the cve estimator this argument
+            is ignored.
+            When the method chosen is "ols" an optimal number of lags is estimated as determined by.
+            When the method is set to "gls" all lags are included.
+        min_length : None or int (optional)
+            Discards tracks shorter than a certain length from the analysis. If `None` (the default)
+            tracks shorter than 3 points if `method == "cve"` or 5 points if `method == "ols" or "gls"`
+            will be discarded.
+        """
+        required_length = (3 if method == "cve" else 5) if min_length is None else min_length
+        filtered_tracks = [track for track in self if len(track) >= required_length]
+        n_discarded = len(self) - len(filtered_tracks)
+
+        if n_discarded:
```

Comment:
Weren't we going to omit the message if `min_length` is specified?

Reply:
no, we still want to emit the warning. If `min_length` isn't specified the value is selected based on the chosen method (`3 if "cve" else 5`) so we still want to notify users if tracks are discarded

Reply:
Doesn't that entirely defeat the purpose of `min_length`? Then you might as well force people to use `filter_tracks` instead and not offer the option at all.

Reply:
sorry, yeah I was confused about this. you're right